pac4j-saml/src/main/java/org/pac4j/saml/client/SAML2Client.java

```diff
@@ -150,24 +150,24 @@ protected void initSAMLContextProvider(final MetadataResolver metadataManager) {
     }
 
     protected MetadataResolver initServiceProviderMetadataResolver(final WebContext context) {
-        this.spMetadataResolver = new SAML2ServiceProviderMetadataResolver(this.configuration.getServiceProviderMetadataPath(),
-                computeFinalCallbackUrl(context),
-                this.configuration.getServiceProviderEntityId(),
+		String finalCallbackUrl = computeFinalCallbackUrl(context);
+		this.spMetadataResolver = new SAML2ServiceProviderMetadataResolver(this.configuration.getServiceProviderMetadataPath(),
+				this.configuration.getServiceProviderMetadataResource(), finalCallbackUrl,
+				(this.configuration.getServiceProviderEntityId().equals("CALLBACK_URL") ? finalCallbackUrl
```

Comment:
I get the idea, but can you remove this for now? Let's try to keep one change at a time, there are already many things to account for.